CONTRIBUTING.md

```diff
@@ -0,0 +1,184 @@
+# Contributing
+
+## General Workflow
+
+1. Fork the repo
+1. Cut a feature branch from develop
+  - make the branch name descriptive
+1. Make commits to your feature branch. Prefix each commit like so:
+  - (feat) Add a new feature
+  - (fix) Fix inconsistent tests
+  - (refactor) ...
+  - (test) ...
+  - (doc) ...
+  - (build) ...
+1. When you've finished with your fix or feature, rebase upstream changes into your branch. submit a [pull request][]
+   directly to develop. Include a description of your changes.
+1. Your pull request will be reviewed by another maintainer. The point of code
+   reviews is to help keep the codebase clean and of high quality and, equally
+   as important, to help you grow as a programmer. If your code reviewer
+   requests you make a change you don't understand, ask them why.
+1. Fix any issues raised by your code reviwer and rebase if appropriate.
+1. Once the pull request has been reviewed, it will be merged by another member of the team. Do not merge your own commits.
+
+## Detailed Workflow
+
+### Fork the repo
+
+Use github’s interface to make a fork of the repo, then add that repo as an upstream remote:
+
+```
+git remote add upstream https://github.com/the-ninja-turtles/turtle
+```
+
+### Cut a feature branch from develop
+
+These commands will help you do this:
+
+``` bash
+# Creates your branch and takes you there
+git checkout -b `your-branch-name` develop
```

Comment:
ah no that's not true, it adds tracking to your branch so that you can just fetch upstream develop and then do git rebase instead of git rebase upstream/develop
I'll remove that to make it clearer